test/functional/wallet_resendwallettransactions.py

```diff
@@ -89,6 +89,10 @@ def run_test(self):
             if txids == [child_txid, txid]:
                 break
             bumped = node.bumpfee(child_txid)
+            # The scheduler queue creates a copy of the added tx after
+            # send/bumpfee and re-adds it to the wallet (undoing the next
+            # removeprunedfunds). So empty the scheduler queue:
+            node.syncwithvalidationinterfacequeue()
             node.removeprunedfunds(child_txid)
             child_txid = bumped["txid"]
```

Comment:
Note to self/reviewer, I think this is why this fix works

Without this, the ordering may be:
1. child1 is created. Added(child1) event is added to queue
2. child2 is created using bumpfee. Removed(child1) and Added(child2) events added to queue
3. removeprunedfunds(child1) deletes child1 from wallet
4. `wallet.transactionAddedToMempool(child1)` actually executes, re-adding child1 to wallet. bug.

And now the ordering must be:
1. child1 is created. Added(child1) event is added to queue
2. `wallet.transactionAddedToMempool(child1)` executes
3. child2 is created using bumpfee. Removed(child1) and Added(child2) events added to queue
4. `wallet.transactionAddedToMempool(child2)` executes, `wallet.transactionRemovedFromMempool(child1)` executes.
5. removeprunedfunds(child1) deletes child1 from wallet